### PR TITLE
Issue #3360: Show CKEditor in the site status report page.

### LINKED ISSUE #3360: [UX] Show CKEditor in the site status report page.
screen shot 2018-11-01 at 11 29 12 pm

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2348

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/3360

### CODE REVIEW

**Comment by klonos** [comment]
Refreshing PR sandbox 😞 😓

File: core/modules/ckeditor/ckeditor.install [hunk 1]
```diff
@@ -13,20 +13,37 @@ function ckeditor_requirements($phase) {
...
+    }
+
+    if (user_access('administer filters')) {
+      $description .= ' ' . $t('To configure text formats, visit the <a href="!url">Text editors and formats</a> configuration page.', array('!url' => url('admin/config/content/formats')));
+    }
+    else {
+      $filter_admins = user_roles(TRUE, 'administer filters');
+      $description .= ' ' . $t('To be able to use CKEditor, you will need to ask a person with one of the following roles to configure the text formats you are allowed to use:');
+      $description .= theme('item_list', array('items' => $filter_admins));
     }
```
Comment (ghost) [inline core/modules/ckeditor/ckeditor.install]: I'm not sure I like this... There was another issue where we removed all 'contact your site administrator' messages. This seems very similar to that.

I'd suggest just displaying the message for all users and if they can't access it they'll contact the person who can.

Comment (jenlampton) [reply core/modules/ckeditor/ckeditor.install]: I'd also recommend removing the whole "else" statement. If they don't have access to administer formats, don't show anything.

Comment (klonos) [reply core/modules/ckeditor/ckeditor.install]: Removed 👍 ... @jenlampton @BWPanda care to review and RTBC this if you're happy with it now? Thanks.

**Comment by klonos** [comment]
Tugboat fail 🤔

File: core/modules/ckeditor/ckeditor.install [hunk 2]
```diff
@@ -13,20 +13,32 @@ function ckeditor_requirements($phase) {
...
+    if (empty($ckeditor_formats)) {
+      $description = $t('CKEditor is installed but not enabled for any text format.');
+    }
+    else {
+      $description = $t('CKEditor is installed and enabled for the following text formats:');
+      $description .= theme('item_list', array('items' => $ckeditor_formats));
     }
+
+    if (user_access('administer filters')) {
+      $description .= ' ' . $t('To configure text formats, visit the <a href="!url">Text editors and formats</a> configuration page.', array('!url' => url('admin/config/content/formats')));
```
Comment (ghost) [inline core/modules/ckeditor/ckeditor.install:33]: What's the reason for using `!url` here (as opposed to `@url`)? The former is more dangerous as no sanitisation is performed. So if `@url` would work just as well, I'd recommend that instead...

```suggestion
      $description .= ' ' . $t('To configure text formats, visit the <a href="@url">Text editors and formats</a> configuration page.', array('@url' => url('admin/config/content/formats')));
```

File: core/modules/system/system.install [hunk 3]
```diff
@@ -54,10 +54,10 @@ function system_requirements($phase) {
   $phpversion = phpversion();
   if (function_exists('phpinfo')) {
     if ($phase == 'runtime') {
-      $php_value = $t('Version: !version (<a href="!infourl">PHP information</a>)', array('!version' => $phpversion, '!infourl' => url('admin/reports/status/php')));
+      $php_value = $t('!version (<a href="!infourl">PHP information</a>)', array('!version' => $phpversion, '!infourl' => url('admin/reports/status/php')));
```
Comment (ghost) [inline core/modules/system/system.install:57]: As above:

```suggestion
      $php_value = $t('@version (<a href="@infourl">PHP information</a>)', array('@version' => $phpversion, '@infourl' => url('admin/reports/status/php')));
```

File: core/modules/system/system.install [hunk 4]
```diff
@@ -54,10 +54,10 @@ function system_requirements($phase) {
...
     }
     else {
-      $php_value = $t('Version: !version', array('!version' => $phpversion));
+      $php_value = $t('!version', array('!version' => $phpversion));
```
Comment (ghost) [inline core/modules/system/system.install:60]: No need for `t()` here as there's nothing to be translated:

```suggestion
      $php_value = $phpversion;
```

File: core/modules/system/system.install [hunk 5]
```diff
@@ -185,7 +185,7 @@ function system_requirements($phase) {
     $class = 'DatabaseTasks_' . Database::getConnection()->driver();
     /* @var DatabaseTasks $tasks */
     $tasks = new $class();
-    $dbinfo = $t('!system version !version', array('!system' => $tasks->name(), '!version' => Database::getConnection()->version()));
+    $dbinfo = $t('!system !version', array('!system' => $tasks->name(), '!version' => Database::getConnection()->version()));
```
Comment (ghost) [inline core/modules/system/system.install:188]: As above:

```suggestion
    $dbinfo = $tasks->name() . ' ' . Database::getConnection()->version();
```

**Review by ghost — CHANGES_REQUESTED** [review]
Few changes needed.

File: core/modules/ckeditor/ckeditor.module [hunk 6]
```diff
@@ -4,7 +4,7 @@
  * Provides integration with the CKEditor WYSIWYG editor.
  */
 
-define('CKEDITOR_VERSION', '4.16.2');
+define('CKEDITOR_VERSION', '4.17.1');
```
Comment (indigoxela) [inline core/modules/ckeditor/ckeditor.module:7]: Touching this is not necessary. The current version of our CKEditor is 4.18.0, BTW.

**Comment by jenlampton** [comment]
Closing in favor of https://github.com/backdrop/backdrop/pull/4509

**Pull request closed without merging**